nautobot/core/api/serializers.py

```diff
@@ -122,13 +127,108 @@ class BaseModelSerializer(OptInFieldsMixin, serializers.HyperlinkedModelSerializ
+            # This is done to ensure smooth CSV export and import processes:
+            #   1. Remove lookup fields that have no instance, e.g., if the Location Parent FK field is missing,
+            #    we remove 'location_parent*' from the CSV.
+            #   2. Replace fields with 'None' values with "NaN" to differentiate between empty and None values during import.
+            all_related_fields_natural_key_lookups = self._get_related_fields_natural_key_field_lookups()
+            case_query = self._build_query_case_for_natural_key_field_lookup(all_related_fields_natural_key_lookups)
+            if isinstance(self.instance, QuerySet):
+                queryset = self.instance
+            else:
+                # We would only need to run one additional query, making this a more efficient method of
+                # obtaining all the natural key values for this instance;
+                queryset = self.Meta.model.objects.filter(pk=self.instance.pk)
+            self.natural_keys_values = queryset.annotate(**case_query).values(
+                *all_related_fields_natural_key_lookups, "pk"
+            )
+
+    def _build_query_case_for_natural_key_field_lookup(self, lookups):
+        """
+        Build a query using Case expressions to handle natural key field instances that do not exist.
+
+        This function constructs a database query with Case expressions to handle natural key lookup fields
+        that may have missing instances. In cases where the natural key field instance does not exist
+        (i.e., is None), this function replaces it with the value 'ObjectNotFound'. This is particularly
+        useful for CSV Export processes, as it allows fields with missing instances to be safely ignored.
+        Such handling is essential for CSV Import processes, as attempting to import missing instances can
+        lead to 'Object Not Found' errors, potentially causing the import to fail.
+
+        Example:
+            Consider a Device model with a related field 'tenant' and a 'name' attribute. In this case, the
+            function can be used as follows:
+
+            Device.objects.annotate(
+                tenant__name=Case(
+                    When(tenant__isnull=False, then=F("tenant__name")),
+                    default="ObjectNotFound"
+                )
+            ).values("tenant__name")
+
+        Explanation:
+            - If `device.tenant` is None, the 'tenant__name' field is set to "ObjectNotFound".
+            - If `device.tenant` is not None, the 'tenant__name' field is set to the actual tenant name.
+
+        Args:
+            lookups: List of natural key lookups
+        """
+        case_query = {}
+        for lookup_field in lookups:
+            # Get the field name of the lookup, e.g the field_name for this lookup `location__parent__parent__name` is `location__parent__parent`
+            *field_name, _ = lookup_field.split("__")
+            field_name = "__".join(field_name)
```

Comment:
So you're really just getting ride of the last field in the lookup expression?

Reply:
Yup